src/map/pc.c

```diff
@@ -10730,19 +10730,20 @@ static int pc_checkitem(struct map_session_data *sd)
 		if (sd->guild && sd->itemcheck & PCCHECKITEM_GSTORAGE) {
 			struct guild_storage *guild_storage = idb_get(gstorage->db,sd->guild->guild_id);
 			if (guild_storage) {
-				for (i = 0; i < MAX_GUILD_STORAGE; i++) {
-					if ((id = guild_storage->items[i].nameid) == 0)
+				for (i = 0; i < guild_storage->items.amount; i++) {
```

Comment:
This would only work if the storage data was compacted, but `guild_storage_delitem()` leaves holes (zeroed entries) on removal. `guild_storage->items.capacity` should be used here instead (and any other iterations should also be checked for this kind of problem)